lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -447,11 +447,13 @@ static bool HasReflectionInfo(ObjectFile *obj_file) {
   return hasReflectionSection;
 }
 
-SwiftLanguageRuntimeImpl::ReflectionContextInterface *
+SwiftLanguageRuntimeImpl::ThreadSafeReflectionContext 
 SwiftLanguageRuntimeImpl::GetReflectionContext() {
-  if (!m_initialized_reflection_ctx)
-    SetupReflection();
-  return m_reflection_ctx.get();
+  // SetupReflection immediately returns if reflection is already setup. We
+  // check that m_initialized_reflection_ctx is set there because it should only
+  // be read after locking the m_add_module_mutex.
+  SetupReflection();
+  return {m_reflection_ctx.get(), m_add_module_mutex};
```

Comment:
Sorry to keep nagging about this, but I still think we should lock the mutex here and then move it into the `ThreadSafeReflectionContext`. Something like this:

```
struct Foo {
    Foo(std::recursive_mutex& m) : guard(m, std::adopt_lock) {}
    std::lock_guard<std::recursive_mutex> guard;
};

Foo foo() {
    std::recursive_mutex m;
    m.lock();
    return Foo(m);
}
```

Reply:
+1 to this outcome.